services/sso/src/main/java/software/amazon/awssdk/services/sso/auth/SsoProfileCredentialsProviderFactory.java

```diff
@@ -109,5 +126,50 @@ public void close() {
             IoUtils.closeQuietly(credentialsProvider, null);
             IoUtils.closeQuietly(ssoClient, null);
         }
+
+        private static String regionFromProfileOrSession(Profile profile, ProfileFile profileFile) {
+            Optional<String> ssoSession = profile.property(ProfileSection.SSO_SESSION.getPropertyKeyName());
+
+            String profileRegion = profile.properties().get(ProfileProperty.SSO_REGION);
+
+            String ssoRegion = ssoSession.isPresent() ?
+                               propertyFromSsoSession(ssoSession.get(), profileFile, ProfileProperty.SSO_REGION) :
+                               profileRegion;
+
+            if (profileRegion != null && !ssoRegion.equalsIgnoreCase(profileRegion)) {
```

Comment:
Should we validate `sso_start_url` as well?

Reply:
I see. I just noticed that the specification mentions validation of the start url, but I may miss some context.

> If the profile and the sso-session both contain sso_region or sso_start_url then SDKs MUST validate that the values in both the profile and sso-session match. If either the sso_region or sso_start_url differ between the profile and sso-session, then SDK MUST raise a validation error.

Reply:
Oops , miss from my side. I read it completely to understand only sso_region is required since ss_start_url is not directly used by sso getRole call.
Corrected this now. Thanks. Good catch.